### DISCUSSION TITLE
UI on Android

### DISCUSSION THREAD

**Discussion by imiron13** [ihhub/fheroes2#6187] [General]
I've noticed you guys added Android support. That's so great. It runs smoothly. And it is just enjoyable to tap with a finger. Surprisingly even though OG was not designed for touch screen interface, most of the UI is convenient as is. Some feedback after the quick testing:
1. Right click. Not supported yet? That is definitely required (monsters check on the World map, etc). I guess it's already on your list. It is possible to implement it using a long press. But it's a bit annoying if you have to tap and wait. Can it be done using two fingers tap? Like tap and hold by one finger anywhere, then tap and remove another finger and it's treated as right click is pressed at a position of the second finger tap until you remove the first finger (or any other method if that's sounds like non-sense)
2. Precise clicks. Most of the UI does not require to aim precisely. One case I noticed that does require precise clicks is an attack with a position selection in the battle screen. This is actually a bit inconvenient even with a mouse. Could it be done like drag n drop from a cell where you want to position your creature to a cell with a creature you want to attack? (for both Touch interface and Mouse interface). Another case is scroll bar: it's it's too narrow, so difficult with a finger, but it's used like only once during game setup, so not really an issue. I saw some games use relative mouse pointing for touch UI (when mouse pointer is not under you finger), it allows to do a very accurate clicks but it's so inconvenient and kills all the fun. 
3. Resolution. On a phone with a rather small screen I found that only 640x480 is playable, higher res makes everything too small and too difficult to tap with your finger. But there is no widescreen resolution for x480. In result - a lot of unused screen area.

This game is in a great shape. Wish you all the best in future improvements.

**Comment by ihhub** [comment]
Hi @imiron13 , all these points are in our TO-DO list. You can check [this issue](https://github.com/ihhub/fheroes2/issues/6008) and [this pull request](https://github.com/ihhub/fheroes2/pull/6092).

**Comment by imiron13** [comment]
Hi @ihhub ,

Wow. You guys three steps ahead. So many discussions about all that already. I should have searched through the issues.

**Comment by Mr-Bajs** [comment]
Been playing some fheroes2 on a tablet. Apart from then things addressed in this thread and https://github.com/ihhub/fheroes2/issues/6008.

Using a 10" tablet with 1280x720 resolution

4. CastIe swipe. I found myself really miss a feature to touch swipe on the screen in the Castle view. A feature to swipe left or right to the next or previous castle. Using the buttons in bottom seams to be to small for easy usage on.

5. Scrollbar buttons. UI was easy enough to be playable and enjoying the game. But the trading resources in the market, the buttons is way small, it was sort of an annoyance to use.
